src/Test.FunctionalTests.Sarif/TestData/Multitool/ValidateCommand/ExpectedOutputs/SARIF2005.ProvideHelpfulToolInformation_Invalid.sarif

```diff
@@ -0,0 +1,102 @@
+{
+  "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+  "version": "2.1.0",
+  "runs": [
+    {
+      "tool": {
+        "driver": {
+          "name": "SARIF Functional Testing",
+          "rules": [
+            {
+              "id": "SARIF2005",
+              "name": "ProvideHelpfulToolInformation",
+              "shortDescription": {
+                "text": "Placeholder."
+              },
+              "fullDescription": {
+                "text": "Placeholder"
+              },
+              "messageStrings": {
+                "Warning_ProvideToolVersion": {
+                  "text": "{0}: Placeholder"
+                },
+                "Warning_ProvideConciseToolName": {
+                  "text": "{0}: Placeholder '{1}' '{2}'"
+                },
+                "Warning_UseNumericToolVersions": {
+                  "text": "{0}: Placeholder '{1}'"
+                }
+              },
+              "helpUri": "http://docs.oasis-open.org/sarif/sarif/v2.1.0/sarif-v2.1.0.html"
+            }
+          ]
+        }
+      },
+      "invocations": [
+        {
+          "executionSuccessful": true
+        }
+      ],
+      "artifacts": [
+        {
+          "location": {
+            "uri": "FunctionalTestOutput.ValidateCommand/Inputs.SARIF2005.ProvideHelpfulToolInformation_Invalid.sarif",
+            "uriBaseId": "TEST_DIR"
+          }
+        }
+      ],
+      "results": [
+        {
+          "ruleId": "SARIF2005",
+          "ruleIndex": 0,
+          "message": {
+            "id": "Warning_ProvideConciseToolName",
```

Comment:
>Warning_ProvideConciseToolName [](start = 19, length = 30)

This should not fire because the name "SARIF functional testing" above is three words. This will go away once you fix your `Split` call.

BUT! Once you have verified that, please change the name to "SARIF functional testing long" to make sure that it _does_ fire when it's more than three words.

Reply:
If you look at the test, i'm adding the "SARIF functional testing long". But, when we get the actual, its changing the sarif to a valid one.

---
In reply to: [444580896](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444580896) [](ancestors = 444580896)

Reply:
I'm not sure what you mean. The input file has "SARIF Functional Testing long". The expected output file (this file) correctly says that it found 4 words instead of the maximum of 3 (lines 56 and 57 below). If the "actual" file matches the "expected output" file, the test would pass. Are you saying that the test fails? If so, can you send me the "actual" file?

---
In reply to: [444582777](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444582777) [](ancestors = 444582777,444580896)

Reply:
what i'm saying is that when the actual contains "SARIF Functional Testing Long", the output tool.driver.name is changed to "SARIF Functional Testing". So, probably, the multitool is solving the issue as well, creating a valid sarif file

---
In reply to: [444604279](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444604279) [](ancestors = 444604279,444582777,444580896)

Reply:
The test populates `tool.driver`. It's the _results_ that tell us whether the analysis was correct.

---
In reply to: [444743290](https://github.com/microsoft/sarif-sdk/pull/1926#discussion_r444743290) [](ancestors = 444743290,444604279,444582777,444580896)